WooCommerce/Classes/POS/Presentation/Card Present Payments/CardPresentPaymentAlertSwiftUIViewModel.swift

```diff
@@ -0,0 +1,24 @@
+import Combine
+import SwiftUI
+
+/// Wraps `CardPresentPaymentAlertViewModel` and provides additional optional view models for custom navigation initiated from the alert in SwiftUI.
+final class CardPresentPaymentAlertSwiftUIViewModel: ObservableObject {
+    let alertViewModel: CardPresentPaymentAlertViewModel
+
+    /// Optional view model to present a WC settings web view if the alert view model conforms to
+    /// `CardPresentPaymentsModalViewModelWCSettingsWebViewPresenting`.
+    @Published var wcSettingsWebViewModel: WCSettingsWebViewModel?
+
+    init(alertViewModel: CardPresentPaymentAlertViewModel) {
+        self.alertViewModel = alertViewModel
+        observeWebViewPresenting()
+    }
+}
+
+private extension CardPresentPaymentAlertSwiftUIViewModel {
+    func observeWebViewPresenting() {
+        let webViewPublisher = (alertViewModel as? CardPresentPaymentsModalViewModelWCSettingsWebViewPresenting)?.webViewModel ??
+        Just<WCSettingsWebViewModel?>(nil).eraseToAnyPublisher()
+        webViewPublisher.assign(to: &$wcSettingsWebViewModel)
+    }
+}
```

Comment:
Nice workaround!

I think we'll be able to make it even better in future – I've realised that we might be able to have `CardReaderConnectionAlertsProviding`/`CardReaderTransactionAlertsProviding` use a generic for the type they provide, and then also have the alerts providers use the same generic. That way we can break the dependency on `CardPresentPaymentsModalViewModel` in the adapted code and just provide our own semantic VMs for each alert. That might be next week though, and it might not work.